### PR TITLE
Update Issue and Pull Request templates

### PR DESCRIPTION
**Pull request by droidmonkey**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Summary
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
Made substantive updates to the issue and pull request templates. Removed unnecessary and often misused sections. 

I removed the checklist section from the pull request template because it really didn't mean anything to anyone. Our robust CI captures the vast majority of issues now and a checklist never stopped anyone from submitting a half-baked PR.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Documentation (non-code change)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -6,44 +6,30 @@ labels: bug
...
-
 ## Context
-[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )
+[NOTE]: # ( Give us any additional information you may have. )
 
 
-## Debug Info
 [NOTE]: # ( Paste debug info from Help → About here )
 KeePassXC - VERSION
 Revision: REVISION
```
Comment (phoerious) [inline .github/ISSUE_TEMPLATE/bug-report.md:35]: I would suggest having at least the OS in there as well or nothing at all. KeePassXC version alone isn't enough in almost all cases, whereas a note of the OS may be sufficient to help some already without repeatedly asking them to provide the full debug info.

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 2]
```diff
@@ -1,25 +1,18 @@
 ---
 name: Feature Request
-about: tell us about a new capability you want to see
+about: tell us about a new feature you want
 title: 
 labels: new feature
```
Comment (phoerious) [inline .github/ISSUE_TEMPLATE/feature-request.md:5]: We should clarify the use of new feature vs wishlist. I've always handled new feature as a tag for PRs and wishlist for something users would like to have (i.e. feature requests). We can rename wishlist to feature request, but eight now we have two tags with unclear connotation.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,16 +1,6 @@
...
-## Type of change
-[NOTE]: # ( Please remove all lines which don't apply. )
-- ✅ Bug fix (non-breaking change which fixes an issue)
-- ✅ Refactor (significant modification to existing code)
-- ✅ New feature (non-breaking change which adds functionality)
-- ✅ Breaking change (fix or feature that would cause existing functionality to change)
-- ✅ Documentation (non-code change)
-
-## Description and Context
+## Summary
```
Comment (phoerious) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: This heading should be removed. Github automatically inserts the commit message at the top, which is basically the summary.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (droidmonkey) [reply .github/ISSUE_TEMPLATE/bug-report.md:35]: Good point, and desktop env for Linux

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 2 above)
Comment (droidmonkey) [reply .github/ISSUE_TEMPLATE/feature-request.md:5]: I prefer one tag, new feature. Now github clearly links issues to PRs in the right hand block.

Comment (phoerious) [reply .github/ISSUE_TEMPLATE/feature-request.md:5]: Then we should get rid of wishlist. Though I still feel like new feature sounds a bit too demanding or definitive to me, whereas feature request wouldn't fit for PRs.

Comment (droidmonkey) [reply .github/ISSUE_TEMPLATE/feature-request.md:5]: If its just a feeling than you can overcome it with some meditation. :wink:

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (droidmonkey) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I wish GitHub would skip over the first heading label or allow for a fill-in tag like `<commitmsg>`

**Comment by droidmonkey** [comment]
I think this meets the mail now. I will replace `wishlist` with `new feature` once merged.

**Comment by droidmonkey** [comment]
Nerd fail

**Comment by droidmonkey** [comment]
Done, will merge after CI.

**Pull request merged**